Detective Trenchcoat buttons and (Det & Curator) Fedora Sprites update

DrDiasyl opened the pull request:
Now every type of det_suit have buttons that can be toggled (Except of those weird disco elysum suits).
Also, now detective and curator fedora uses the normal fedora sprite which is way better then the old one. Also there is now a noir fedora so detective can now wear epic fedora without losing armor.

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

New varient for detectives to wear their funny trenchcoats (Or noir suit coat).
Detective's who want to wear noir fedora no longer need to lack their head protection.
Detective and Curator fedora sprites now have better sprites which is taken from basic grey fedora sprite.

[Brown No Buttons](https://cdn.discordapp.com/attachments/528115508080541709/933392775704117358/Det_Brown.png)
[Brown Buttons](https://cdn.discordapp.com/attachments/528115508080541709/933392974719639572/Det_Brown_Buttons.png)
[Grey No Buttons](https://cdn.discordapp.com/attachments/528115508080541709/933392776169652294/Det_Grey.png)
[Grey Buttons](https://cdn.discordapp.com/attachments/528115508080541709/933392974941941780/Det_Grey_Buttons.png)
[Noir No Buttons](https://cdn.discordapp.com/attachments/528115508080541709/933392776970797126/Det_Noir.png)
[Noir Buttons](https://cdn.discordapp.com/attachments/528115508080541709/933392777214038026/Det_Noir_Buttons.png)

**And also in those screenshots you can see how new brown det fedora looks like.**
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Some detective's want to wear it button closed or button open, can add variety to their style.
Now grey fedora for noir style no longer lacks protection (The normal grey fedora still stays, there is just new noir fedora with same sprite)
Better det and curator fedora sprites make it more pleasant to look at (Especially up close)
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: DrTuxedo (DrDiasyl)
expansion: Nanotrasen have stitched buttons to the detective issue trenchcoats (and noir coats), making it possible to toggle close or open them!
fix: Nanotrasen finally started making detective noir fedoras from the same materials as their brown counterparts, making them have same protection levels!
imageadd: New detective and curator fedora sprites that are more pleasent looking.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

DrDiasyl committed:
Trenchcoat buttons and misc

Now every type of det_suit have buttons that can be toggled (Except of those weird disco elysum suits).
Also, now detective and curator fedora uses the normal fedora sprite which is way better then the old one. Also there is now a noit fedora so detective can now wear epic fedora without losing armor.

Valtosin commented:
Finally, detective can cosplay exhibitionist

DrDiasyl commented:
> Finally, detective can cosplay exhibitionist

Maybe he is just cold.

ZeWaka commented on code/modules/clothing/head/jobs.dm:
```diff
@@ -117,6 +117,10 @@
 	else
 		to_chat(user, span_warning("You just took a candy corn! You should wait a couple minutes, lest you burn through your stash."))
 
+//Now if detective wants to get a noir fedora it will be too protected.
```
Comment is unclear - what does `it will be too protected.` mean?

DrDiasyl replied:
Before it detective had basic gray fedora in their wardrobe which didn't had protection, and I added a second gray fedora but with protection (For detectives only)

ZeWaka commented on code/modules/clothing/suits/jobs.dm:
```diff
@@ -236,3 +236,18 @@
 	icon_state = "jamrock_blazer"
 	inhand_icon_state = "jamrock_blazer"
 
+//This kind of shenanigans exist because I have no other idea how to make those disco suits not togglable.
+/obj/item/clothing/suit/det_suit
+	name = "old trenchcoat"
+	desc = "Looks like mapper fucked up and placed wrong trenchcoat."
```
Why does this type exist if you're replacing it?

DrDiasyl replied:
I don't want to touch Disco Elysium coats, they're too cool to have buttons, so I had to leave this so those can still exist without buttons.

Fikou replied:
then replace this type with disco elysium coat

DrDiasyl replied:
> then replace this type with disco elysium coat

hmmmm

DrDiasyl replied:
Done and fixed with the last commit. @Fikou @ZeWaka

DrDiasyl commented:
Also, what you guys think about adding a fully open grey trenchcoat variant? So assistants now can TRULY flash at people lmao.

DrDiasyl committed:
Made noir fedora comment more clear

JohnFulpWillard commented:
Are you sure your branch is updated? I remember the /toggle/ subtype being removed at some point

DrDiasyl commented:
> Are you sure your branch is updated? I remember the /toggle/ subtype being removed at some point
> 
> Edit: #61941

I have downloaded codebase literally yesterday

DrDiasyl commented:
> Are you sure your branch is updated? I remember the /toggle/ subtype being removed at some point
> 
> Edit: #61941

Also bruh, suit/toggle still exists and is used in labcoats.

EricZilla commented:
Looking forwards to that grey fedora, but the most important question, does it still got candy corn

DrDiasyl commented:
**Don't merge this, I have realised that sprites didn't committed somehow**

DrDiasyl commented:
Yo what, I got a dislike from oranges

DrDiasyl committed:
Actually commits sprites

Actually commits sprites of fedora's and trenchcoats, for some reason it didn't got commited before.

DrDiasyl commented:
**Now it must be fine**

Krysonism reviewed (approved):
Although these trenchcoats could use a full resprite, the fedora and buttoning changes are good.

DrDiasyl commented:
Now waiting for this to get merged.

DrDiasyl committed:
Makes a disco det_suits their own type

This is to prevent shenanigans with types, due to not wanting to make those toggleable

DrDiasyl commented:
Wait, checks failed??

DrDiasyl committed:
Fixes check fails

I hope it will fix them

DrDiasyl commented:
Hell

DrDiasyl committed:
Checks fix again (Bruh)

I made a typo so it didn't work bruh

DrDiasyl commented:
WHAT THE FUCK IS WRONG WITH YOU? I DIDN'T EVEN TOUCH YOU.
![image](https://user-images.githubusercontent.com/42353186/150651745-17770209-9ccf-42a5-956c-80f34dfde8e1.png)
![image](https://user-images.githubusercontent.com/42353186/150651746-e17a96d9-e9f1-4398-8132-92ea4135dc98.png)

DrDiasyl commented:
nvm, it's not even a error lol

DrDiasyl committed:
(Probably finally) fixes the bullshit error

Deleted ""unique trenchcoat and fedora"" from detective's closet in Deltastation. I will not fucking make curator jacket togglable for this variable changed shit on Deltastation which is not even meet anywhere else!

DrDiasyl commented:
Fucking fixed

DrDiasyl commented:
Ahhhhh, can someone fix this for me?

Imaginos16 commented:
@DrDiasyl This PR has been inactive for a while. Are you still working on it or taking a break?

DrDiasyl commented:
> @DrDiasyl This PR has been inactive for a while. Are you still working on it or taking a break?

Well, I think today I will fix the merge conflict..

DrDiasyl commented:
Okay I'm lazy ass, will fix tomorrow, I swear

DrDiasyl committed:
Trying to fix this piece of bullshit merge conflict part 1

DrDiasyl committed:
Fixing that bullshit merge conflict part 2

Imaginos16 commented:
Closing it at request of author.
